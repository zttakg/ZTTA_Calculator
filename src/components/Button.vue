<script>
export default {
  name: "Button",

  props: {
    type: {
      type: String,
      default: "button"
    },
    color: String,
    to: {
      type: String,
      default: null
    },
    size: {
      type: String,
      default: "medium"
    },
    theme: String,
    block: Boolean,
    disabled: Boolean,
    icon: Boolean,
    value: null
  },

  computed: {
    classes() {
      /* eslint-disable indent */
      const themes = this.theme
        ? this.theme
            .split(" ")
            .filter(t => t !== "")
            .map(t => `button_style_${t}`)
        : [];
      /* eslint-enable indent */

      const classes = [
        "button",
        {
          [`button_color_${this.color}`]: this.color,
          [`button_size_${this.size}`]: this.size,
          button_block: this.block,
          button_disabled: this.disabled,
          button_icon: this.icon
        },
        ...themes
      ];

      return classes;
    }
  },

  methods: {
    handleClick(event) {
      this.$emit("click", event);
    }
  },

  // eslint-disable-next-line no-unused-vars
  render(h) {
    const ButtonTag = this.to ? "a" : "button";
    const type = this.to ? null : this.type;

    return (
      <ButtonTag
        onClick={this.handleClick}
        class={this.classes}
        type={type}
        href={this.to}
      >
        {this.$slots.default}
      </ButtonTag>
    );
  }
};
</script>
