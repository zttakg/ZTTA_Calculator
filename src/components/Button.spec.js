import { shallowMount } from "@vue/test-utils";
import Button from "./Button";

describe("<Button /> component", () => {
  it("should render correct without props", () => {
    const wrapper = shallowMount(Button);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render all props as link", () => {
    const wrapper = shallowMount(Button, {
      propsData: {
        to: "#",
        type: "submit",
        color: "blue",
        size: "xxxl",
        theme: "solid rounded",
        block: true,
        disabled: true,
        icon: true
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render all props as button", () => {
    const wrapper = shallowMount(Button, {
      propsData: {
        type: "submit",
        color: "blue",
        size: "xxxl",
        theme: "solid rounded",
        block: true,
        disabled: true,
        icon: true
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
