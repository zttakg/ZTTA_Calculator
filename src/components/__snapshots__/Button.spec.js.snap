// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> component should render all props as button 1`] = `<button type="submit" class="button button_color_blue button_size_xxxl button_block button_disabled button_icon button_style_solid button_style_rounded"></button>`;

exports[`<Button /> component should render all props as link 1`] = `<a href="#" class="button button_color_blue button_size_xxxl button_block button_disabled button_icon button_style_solid button_style_rounded"></a>`;

exports[`<Button /> component should render correct without props 1`] = `<button type="button" class="button button_size_medium"></button>`;
